www/source/docs/best-practices.html.md.erb

```diff
@@ -303,35 +303,186 @@ Chef Habitat packages may be exported with the Supervisor directly into a [a var
+- The habitat http-gateway (9631/TCP) listener
+- the service name becomes available in DNS (as `hab-bastion` or `hab-bastion.namespace-name`, etc) so any pod can find it
 
-```bash
+```yaml
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: hab-bastion
+spec:
+    spec:
+      securityContext:
+        fsGroup: 42
+```
+
+This section sets a group ownership for the persistent volume mount point, so that the Habitat supervisor can write to it.  The Habiat user (`hab`) by default has uid `42` and gid `42`
+
+```yaml
+      containers:
+      - name: hab-bastion
+        image: mydockerorigin/hab_bastion:latest
+        args:
+        - '--permanent-peer'
+```
+
+The `image:` line defines you where to get the docker container from. In this case the image is created from a Chef Habitat plan using the `hab pkg export docker` functionality. It runs no service except the Chef Habitat supervisor (hab-sup).
+The argument `--permanent-peer` instructs the supervisor to act as a permanent peer.
+
+```yaml
+        resources:
+          requests:
+            memory: "100Mi"
+            cpu: "100m" # equivalent to 0.1 of a CPU core
+```
+
+Resource requests are important to inform the kubernetes scheduler - without them, some nodes could be overloaded while others are underutilized.
+
+```yaml
+        ports:
+        - name: gossip-listener
+          protocol: UDP
+          containerPort: 9638
+        - name: http-gateway
+          protocol: TCP
+          containerPort: 9631
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 9631
+          initialDelaySeconds: 5
+          periodSeconds: 10
+        livenessProbe:
+          httpGet:
+            path: /
+            port: 9631
+          initialDelaySeconds: 15
+          periodSeconds: 20
+```
+
+The livenessProbe lets Kubernetes know if the pod is healthy or not.  If not it will get restarted.
+The readinessProbe signals to Kubernetes that the pod has started up successfully.
+
+```yaml
+        volumeMounts:
+        - name: hab-bastion
+          mountPath: /hab/sup
+  volumeClaimTemplates:
+  - metadata:
+      name: hab-bastion
+    spec:
+      accessModes: [ "ReadWriteOnce" ]
+      # uncomment if you don't have a default storageclass
+      # storageClassName: "standard"
+      resources:
+        requests:
+          storage: 10Gi
+```
+
+All of the Habitat supervisor's state data is stored under `/hab/sup` - we mount this on a persistent volume so it gets re-attached if the pod is ever relaunched. The data persists!
+
+### Create a Kubernetes Deployment that works with the Bastion
+
+The following is an example Kubernetes Stateful Set built from the CockroachDB plan. To illustrate the Bastion pattern, the `--peer hab-bastion` arguments are the part of the configuration where these pods are instructed how to reach the bastion, using the aforementioned service `hab-bastion` as a DNS-resolvable host name.
```

Comment:
```suggestion
The following is an example of Kubernetes "Stateful Set" built from the CockroachDB plan. The Bastion pattern uses the `--peer hab-bastion` configuration arguments to instruct the Kubernets pods to use the `hab-bastion` service as a DNS-resolvable host name.
```

Reply:
whoops a couple typos on this suggestion @kagarmoe.  "an example of Kubernetes"  and "Kubernets"